src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -21,120 +103,808 @@ const ActorInit En_Pamera_InitVars = {
+            return 1;
+        }
+        actorPtr = actorPtr->next;
+    }
+    this->actor.child = NULL;
+    return 0;
+}
+
+void func_80BD8588(EnPamera* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[((this->actor.params & 0xFF0) >> 4)];
+    Vec3f sp28;
+
+    if (path == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    if (gSaveContext.weekEventReg[61] & 4) {
+        path = &globalCtx->setupPathList[(u8)path->unk1];
+    }
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+    this->pathIndex = 0;
+    this->pathPointsCount = path->count;
+    this->animatedTextures = (u8)path->unk1;
+    Math_Vec3s_ToVec3f(&sp28, this->pathPoints);
+    this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp28);
+}
+
+void func_80BD8658(EnPamera* this) {
+    s16 nextCutscene;
+    s32 i;
+
+    nextCutscene = this->actor.cutscene;
+    for (i = 0; i < 2; i++) {
+        if (nextCutscene == -1) {
+            break;
+        }
+        this->cutscenes[i] = nextCutscene;
+        nextCutscene = ActorCutscene_GetAdditionalCutscene(nextCutscene);
+    }
+}
+
+void EnPamera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPamera* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BD8700(EnPamera* this) {
+    this->unk_31C = 0;
+    this->actor.flags &= -2;
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BD8758;
+}
+
+void func_80BD8758(EnPamera* this, GlobalContext* globalCtx) {
+    s32 temp = (this->unk_31C < 0x709) ^ 1;
+
+    this->unk_31C++;
+    if (temp) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscenes[0]) && (this->cutscenes[0] != -1)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutscenes[0], &this->actor);
+            func_800E02AC(Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(this->cutscenes[0])), this);
+            this->actor.speedXZ = 1.5f;
+            func_800BDC5C(&this->skelAnime, animations, 1);
+            this->actor.shape.rot.y = this->actor.home.rot.y;
+            this->actor.world.rot.y = this->actor.home.rot.y;
+            func_80BD9338(this, globalCtx);
+            func_80BD8908(this);
+        } else {
+            if ((this->cutscenes[0] != -1) && (this->actor.xzDistToPlayer < 1000.0f)) {
+                ActorCutscene_SetIntentToPlay(this->cutscenes[0]);
+            } else {
+                this->actor.speedXZ = 1.5f;
+                func_800BDC5C(&this->skelAnime, animations, 1);
+                this->actor.shape.rot.y = this->actor.home.rot.y;
+                this->actor.world.rot.y = this->actor.home.rot.y;
+                func_80BD9338(this, globalCtx);
+                func_80BD8908(this);
+            }
+        }
+    }
+    if (this->actor.draw && (this->actionFunc != func_80BD8964)) {
+        if ((this->actor.child) && (this->actor.child->id == 5) && (this->actor.child->world.rot.y == 0)) {
```

Comment:
```suggestion
    if ((this->actor.draw != NULL) && (this->actionFunc != func_80BD8964)) {
        if ((this->actor.child != NULL) && (this->actor.child->id == 5) && (this->actor.child->world.rot.y == 0)) {
```
Since `this->actor.draw` and `this->actor.child` are pointers, I'd prefer an explicit check to not equal `NULL`.

Reply:
Also looks like the id 5 would be `ACTOR_EN_DOOR`